### ISSUE TITLE
ERROR: Multiple .bb files are due to be built which each provide virtual/snes9x 

### ISSUE THREAD

**Issue by graugans** [sergioprado/meta-game-emulators#3]
When I try to build this layer against my [UDOO Board](https://github.com/graugans/meta-fsl-arm-extra)  
layer I get the following error message:

```
ERROR: Multiple .bb files are due to be built which each provide virtual/snes9x (/data/FSL/fsl-community-bsp-master/sources/meta-game-emulators/recipes-snes/snes9x/snes9x_1.53.bb /data/FSL/fsl-community-bsp-master/sources/meta-game-emulators/recipes-snes/snes9x-sdl/snes9x-sdl_1.53.bb).
 This usually means one provides something the other doesn't and should.
```

Even when I add the following to my machine conf:

```
PREFERRED_PROVIDER_virtual/snes9x ?= "snes9x-sdl"
```

**Comment by sergioprado** [comment]
Hello Christian,

Thanks for notifying me of the issue.

What recipe are you trying to process? Would you please attach your local.conf?

**Comment by graugans** [comment]
Sorry for the late reply. The problem seems to be caused by the recipe [core-image-emulators-full.bb](https://github.com/sergioprado/meta-game-emulators/blob/master/recipes-tests/images/core-image-emulators-full.bb#L9) because you depend on both variants of snes9x. 

My machine config looks like this:

```
MACHINE ??= 'udooqdl'
DISTRO ?= 'poky-emulators'
PREFERRED_PROVIDER_virtual/snes9x = "snes9x"
PACKAGE_CLASSES ?= "package_ipk"
EXTRA_IMAGE_FEATURES = "debug-tweaks package-management"
USER_CLASSES ?= "buildstats image-mklibs"
PATCHRESOLVE = "noop"
BB_DISKMON_DIRS = "\
    STOPTASKS,${TMPDIR},1G,100K \
    STOPTASKS,${DL_DIR},1G,100K \
    STOPTASKS,${SSTATE_DIR},1G,100K \
    STOPTASKS,/tmp,100M,100K \
    ABORT,${TMPDIR},100M,1K \
    ABORT,${DL_DIR},100M,1K \
    ABORT,${SSTATE_DIR},100M,1K \
    ABORT,/tmp,10M,1K"
PACKAGECONFIG_append_pn-qemu-native = " sdl"
PACKAGECONFIG_append_pn-nativesdk-qemu = " sdl"
ASSUME_PROVIDED += "libsdl-native"
CONF_VERSION = "1"

DL_DIR ?= "${BSPDIR}/downloads/"
ACCEPT_FSL_EULA = "1"
```

**Comment by sergioprado** [comment]
Thanks for the feedback.

Since I wanted to include multiple variants of the snes9x in the same image, I removed the virtual provider declarations.

The problem was fixed on commit 915d647e11177a9712cf2991e8592a1d21390686.

Please let me know if it is working now.

**Comment by graugans** [comment]
I had to fix some URLs due to the fact that the projects moved or the hoster was gone. You can find my updates [here](https://github.com/graugans/meta-game-emulators)

**Comment by sergioprado** [comment]
Hello Christian.

Would you mind to send me a pull request of your changes?

Thanks,

Sergio Prado
Embedded Labworks
Office: +55 11 2628-3461
Mobile: +55 11 97123-3420

2016-01-04 17:27 GMT-02:00 Christian Ege <notifications@github.com>:

**Comment by sergioprado** [comment]
Would you mind to send me a pull request of your changes?

**Comment by graugans** [comment]
Sure, I'll prepare one in the next few days.

**Comment by sergioprado** [comment]
Fixed on commit 915d647.
